lib/irb/cmd/edit.rb

```diff
@@ -0,0 +1,66 @@
+require_relative "nop"
+
+module IRB
+  # :stopdoc:
+
+  module ExtendCommand
+    class Edit < Nop
+      class << self
+        def transform_args(args)
+          # Return a string literal as is for backward compatibility
+          if args.nil? || args.empty? || string_literal?(args)
+            args
+          else # Otherwise, consider the input as a String for convenience
+            args.strip.dump
+          end
+        end
+
+        private
+
+        def string_literal?(args)
+          sexp = Ripper.sexp(args)
+          sexp && sexp.size == 2 && sexp.last&.first&.first == :string_literal
+        end
+      end
+
+      def execute(*args)
+        path = args.first
+
+        if path.nil? && (irb_path = @irb_context.irb_path)
+          path = irb_path
+        end
+
+        if !File.exist?(path)
+          require_relative "show_source"
+          show_source_cmd = ShowSource.new(@irb_context)
+
+          source =
+            begin
+              show_source_cmd.send(:find_source, path)
```

Comment:
Can we change `find_source` to a public singleton method that also takes `irb_context` to clarify that it's now a method shared with other classes? It's hard to notice that the method is shared if it's kept private.

Reply:
Updated